### PR TITLE
[MINOR][ML] Fixed MLlib build warnings

### PR DESCRIPTION
**Pull request by jkbradley**
## What changes were proposed in this pull request?

Fixes to eliminate warnings during package and doc builds.
## How was this patch tested?

Existing unit tests

### CODE REVIEW

**Comment by jkbradley** [comment]
Previous warnings:
```
[info] Compiling 148 Scala sources and 60 Java sources to /Users/josephkb/spark/mllib/target/scala-2.11/test-classes...
[warn] /Users/josephkb/spark/mllib/src/test/scala/org/apache/spark/ml/tree/impl/RandomForestSuite.scala:328: match may not be exhaustive.
[warn] It would fail on the following input: ContinuousSplit()
[warn]       case n: InternalNode => n.split match {
[warn]                                 ^
[warn] /Users/josephkb/spark/mllib/src/test/scala/org/apache/spark/ml/tree/impl/RandomForestSuite.scala:327: match may not be exhaustive.
[warn] It would fail on the following input: LeafNode()
[warn]     model.rootNode match {
[warn]           ^
[warn] /Users/josephkb/spark/mllib/src/test/scala/org/apache/spark/ml/tree/impl/RandomForestSuite.scala:351: match may not be exhaustive.
[warn] It would fail on the following input: LeafNode()
[warn]     def getChildren(rootNode: Node): Array[InternalNode] = rootNode match {
[warn]                                                            ^
[warn] three warnings found


 /Users/josephkb/spark/examples/src/main/scala/org/apache/spark/examples/mllib/AbstractParams.scala:41: method declarations in class TypeApi is deprecated: Use `decls` instead
[warn]     val allAccessors = tpe.declarations.collect {
[warn]                            ^
[warn] 82 warnings found
[info] Note: /Users/josephkb/spark/examples/src/main/java/org/apache/spark/examples/mllib/JavaStratifiedSamplingExample.java uses unchecked or unsafe operations.
[info] Note: Recompile with -Xlint:unchecked for details.
[warn] /Users/josephkb/spark/examples/src/main/scala/org/apache/spark/examples/mllib/AbstractParams.scala:41: method declarations in class TypeApi is deprecated: Use `decls` instead
[warn]     val allAccessors = tpe.declarations.collect {
[warn] 
```

File: examples/src/main/scala/org/apache/spark/examples/mllib/AbstractParams.scala [hunk 1]
```diff
@@ -38,7 +38,7 @@ abstract class AbstractParams[T: TypeTag] {
    */
   override def toString: String = {
     val tpe = tag.tpe
-    val allAccessors = tpe.declarations.collect {
+    val allAccessors = tpe.decls.collect {
```
Comment (srowen) [inline examples/src/main/scala/org/apache/spark/examples/mllib/AbstractParams.scala]: I'm 80% sure we tried to change this and it fails to compile in 2.10.

Comment (jkbradley) [inline examples/src/main/scala/org/apache/spark/examples/mllib/AbstractParams.scala]: Oh OK.  Was there a decision not to suppress the warning here?

Comment (srowen) [inline examples/src/main/scala/org/apache/spark/examples/mllib/AbstractParams.scala]: Hm, not sure why... somehow remember that I couldn't figure out how to suppress it. There are a few like this that can be suppressed, yes.

Comment (holdenk) [inline examples/src/main/scala/org/apache/spark/examples/mllib/AbstractParams.scala]: Yes, so I also took a look at this back when I was doing my initial cleanup, there isn't an easy way to suppress deprecation warnings in Scala like these ones (the answer from the Scala people seem to be either "you can't" or  "you could write a compiler plugin" which isn't a feasible option).

We should definitely do this once we get rid of Scala 2.10 (I created https://issues.apache.org/jira/browse/SPARK-14417 to remind us to do the cleanup once 2.10 is dead).

Personally I think the opportunity to get cleanup using these deprecated APIs without version specific code is maybe part of a good reason why we should consider removing 2.10 support, but that is being discussed more on the mailing list.

Comment (jkbradley) [inline examples/src/main/scala/org/apache/spark/examples/mllib/AbstractParams.scala]: OK thanks!  I reverted this change.

**Pull request closed without merging**